onnx/mapping.py

```diff
@@ -28,7 +28,12 @@
     int(TensorProto.STRING): np.dtype('object')
 }
 
-NP_TYPE_TO_TENSOR_TYPE = {v: k for k, v in TENSOR_TYPE_TO_NP_TYPE.items()}
+NP_TYPE_TO_TENSOR_TYPE = {}
+for tensor_type, numpy_type in TENSOR_TYPE_TO_NP_TYPE.items():
+    # Currently native numpy does not support bfloat16 so TensorProto.BFLOAT16 is ignored for now
+    # Numpy float16 array is only reversed to TensorProto.FLOAT16
+    if tensor_type != TensorProto.BFLOAT16:
+        NP_TYPE_TO_TENSOR_TYPE[numpy_type] = tensor_type
```

Comment:
minor nit/suggestion: I think you can just add the if condition to the original construct as below:
```py
   NP_TYPE_TO_TENSOR_TYPE = {v: k for k, v in TENSOR_TYPE_TO_NP_TYPE.items() if k != TensorProto.BFLOAT16}
```

Reply:
Looks good... I will update in another PR: https://github.com/onnx/onnx/pull/3674 Thanks!